src/main/java/com/jcraft/jsch/ChannelSftp.java

```diff
@@ -643,6 +653,32 @@ public void _put(InputStream src, String dst, SftpProgressMonitor monitor, int m
         throw (SftpException) e;
       throw new SftpException(SSH_FX_FAILURE, e.toString(), e);
     }
+  
+  public boolean checkChunkFailed(InputStream src, String dst, String ChunkStart)throws SftpException{
+	  try{
+	      ((MyPipedInputStream)io_in).updateReadSide();
+
+	      byte[] dstb=Util.str2byte(dst, fEncoding);
+	      long skip=0;
+	      try{
+		  SftpATTRS attr=_stat(dstb);
+		  skip=attr.getSize();
+	      }
+	      catch(Exception eee){
+		  //System.err.println(eee);
+	      }
+	      
+	      if(skip>0){
+	    	  if(skip == Long.parseLong(ChunkStart))
+	    		  return false;
+	    	  else return true;
+	      }
+	      else return false;
+	  }
+	  catch(Exception e){
+		  throw new SftpException(SSH_FX_FAILURE, "Cannot check-" + e.toString());
```

Comment:
1. We should make sure `e` isn't already an instance of `SftpException` first, and if it is, just rethrow it directly.
2. We should also add the original exception as the cause of the new `SftpException`.

```
if (e instanceof SftpException)
  throw (SftpException) e;
throw new SftpException(SSH_FX_FAILURE, e.toString(), e);
```